### ISSUE TITLE
Consumes and Produces attributes are ignored in ASP .NET Core Web API

### ISSUE THREAD

**Issue by caioproiete** [RicoSuter/NSwag#2421]
I'm using [NSwag.AspNetCore v13.0.6](https://www.nuget.org/packages/NSwag.AspNetCore/13.0.6) (latest version as of this writing), and NSwag seem to be ignoring the `Consumes` and `Produces` attributes.

Am I missing anything?

```csharp
[HttpPost("/api/stuff")]
[Consumes("text/plain")]
[Produces("text/plain")]
public IActionResult PostStuff()
{
    return Ok();
}
```

![image](https://user-images.githubusercontent.com/177608/65561338-6a787880-df18-11e9-8c54-395934569258.png)

_Relates to #420_

**Comment by RicoSuter** [comment]
You also need the formatters:

https://github.com/RicoSuter/NSwag/issues/420#issuecomment-467935882

**Comment by praveena-m** [comment]
I do have the same problem.
![image](https://user-images.githubusercontent.com/1097942/65852832-04a34b00-e375-11e9-903a-a740e416a320.png)

**Comment by ptr1120** [comment]
the same problem on my side, ConsumesAttribute completely ignored.

**Comment by RicoSuter** [comment]
Did you read the comment? Just adding [Consumes] won't work as ASP.NET Core also needs a formatter...

**Comment by RicoSuter** [comment]
Where/how should we document this? Maybe we also need a sample etc. (there are unit tests for that however)

**Comment by ptr1120** [comment]
Sorry, thanks. Problem was that the link in #420 to https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs#L426-L461 does not work anymore.

**Comment by ThomasBaechlerConti** [comment]
I am still having this issue. Since this issue claims that everything is correct, I investigated further.

I have an operation that is supposed to consume application/octet-stream, and I have a matching input processor.

```c#
[HttpPost]
[Consumes("application/octet-stream")]
public string EndpointConsumesBinaryData([FromBody] byte[] data)
{
    return string.Empty;
}
```

```c#
internal sealed class BinaryInputFormatter : InputFormatter
{
    public BinaryInputFormatter()
    {
        SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("application/octet-stream"));
    }

    public override async Task<InputFormatterResult> ReadRequestBodyAsync(InputFormatterContext context)
    {
        using var mem = new MemoryStream();
        await context.HttpContext.Request.Body.CopyToAsync(mem);
        return InputFormatterResult.Success(mem.ToArray());
    }

    protected override bool CanReadType(Type type)
    {
        return type.IsAssignableFrom(typeof(byte[]));
    }
}
```

I registered this input processor and the ASP.NET Core API explorer lists the operation correctly. Even NSwag seems to pick up the Consumes attribute. I created a Document Processor and put a breakpoint in the Process method.

Looking at `DocumentProcessorContext.Document.Operations[0].Operation`, I see that the `ActualConsumes` property contains "application/octect-stream", but the `RequestBody.Content` dictionary only contains an "application/json" content.

So, in conclusion, NSwag does ignore the consumes attribute and seems to always assume application/json.

**Comment by Maxwellwr** [comment]
I think the problem is in this code: https://github.com/RicoSuter/NSwag/blob/master/src/NSwag.Core/OpenApiOperation.cs#L236

**Comment by jvmlet** [comment]
here is the culprit https://github.com/RicoSuter/NSwag/blob/4e94b9d2bd558eb7bc2fa5f944b6a3ff08d62c68/src/NSwag.Core/OpenApiResponse.cs#L73-L86

**Comment by MKGNZ** [comment]
Any update to this issue? still open?
